src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -0,0 +1,192 @@
+    //pull out the records, but don't reify them into workflows yet as we need their submission ID
+    val wfRecsFuture: Future[Seq[WorkflowRecord]] = datasource.inTransaction { dataAccess =>
+      dataAccess.workflowQuery.findWorkflowByIds(workflowIds).result
+    }
+
+    //this terrifying monster constructs the workflow options and the list of reified workflows
+    val submitThingsFuture = for {
+      wfRecs <- wfRecsFuture //get the records
+      assert(wfRecs.forall( _.submissionId == wfRecs.head.submissionId)) //sanity check they're all in the same submission
+      execSvcWfOpts <- buildWorkflowOptions(wfRecs.head.submissionId) //build the workflow opts
+      unpackedWfs <- loadWorkflows(wfRecs) //reify to real workflows
+      assert(unpackedWfs.size == wfRecs.size) //there should be the right number of them
+      assert(unpackedWfs.forall(_.isDefined)) //and they should all have been found
+      toSubmit = unpackedWfs.map(_.get) //un-option them too
+    } yield {
+      (execSvcWfOpts, toSubmit)
+    }
+
+    //now we have a future of workflow options and the workflows themselves...
+    submitThingsFuture map {
+      case (wfOpts: ExecutionServiceWorkflowOptions, unpackedWfs: Seq[Workflow]) =>
+        //make the list of workflow inputs
+        val wfInputsBatch = unpackedWfs map { wf =>
+          val methodProps = wf.inputResolutions map {
+            case svv: SubmissionValidationValue if svv.value.isDefined =>
+              svv.inputName -> svv.value.get
+          }
+          MethodConfigResolver.propertiesToWdlInputs(methodProps.toMap)
+        }
+        //why is toJson going red? it's right in this import...
+        import ExecutionJsonSupport.ExecutionServiceWorkflowOptionsFormat
+        executionServiceDAO.submitWorkflows("WHERE'S THE WDL?", wfInputsBatch, Option(wfOpts.toJson.toString), getUserInfo(credential))
+
+        //TODO: if cromwell submission fails, set workflow to Failed rather than WorkflowFailure
+
+
+    }
+    LookForWorkflows
+  }
+
+  //Returns a tuple of some weird things we need later.
+  //these are: the user associated with the submission, the submission's workspace, and its billing project
+  def getSomeThings(submissionId: UUID): Future[(Option[RawlsUser], Option[WorkspaceRecord], RawlsBillingProject)] = {
+    datasource.inTransaction { dataAccess =>
+      for {
+        subRec <- dataAccess.submissionQuery.findById(submissionId).result
+        userOpt <- dataAccess.rawlsUserQuery.load(RawlsUserRef(RawlsUserSubjectId(subRec.head.submitterId)))
+        wsOpt <- dataAccess.workspaceQuery.findByIdQuery(subRec.head.workspaceId).result
+        workspace <- wsOpt
+        billingProject <- dataAccess.rawlsBillingProjectQuery.load(RawlsBillingProjectName(workspace.namespace))
+      } yield {
+        (userOpt, wsOpt.headOption, billingProject)
+      }
+    }
+  }
+  
+  def buildWorkflowOptions(submissionId: UUID): Future[ExecutionServiceWorkflowOptions] = {
```

Comment:
Yeah - no WorkspaceContext or UserInfo any more means I have to jump through hoops to reconstruct them from the things I have. The workspace is used both for the bucket name and to look up the billing project, and I ended up with something weird like:

```
val workspaceFuture = blah
workspaceFuture.map { //to get the workspace name }
workspaceFuture.map { //again to get the billing project }
```

Mapping over the same future multiple times felt weird to me.

Jeff also pointed out the monad-mixing thing, so I suspect this might get nuked in favour of something else anyway. Not entirely sure what though.